feat(flags): Add generic webhook provider (#82263)
getsentry/sentry: src/sentry/flags/docs/api.md

## Webhooks [/organizations/<organization_id_or_slug>/flags/hooks/provider/<provider>/]

### Create Generic Flag Log [POST]

> A flag log event must be emitted after every flag definition change which influences a flag's evaluation. Updates to a flag's internal metadata that does not change a flag's evaluation logic do not need to be emitted to Sentry's audit-log. We are only concerned with changes which could have influenced behavior.

Reviewer: ```suggestion
A flag log event must be emitted after every change which influences a flag's evaluation. Updates to a flag's metadata that do not change its evaluation logic do not need to be emitted to this endpoint.
```

> Sentry does not currently have a concept of disambiguating flag changes by project and environment. Everything is done at the organization level. Flag's which change for multiple environment or projects or whatever internal grouping exists within the generic provider must be de-duplicated. To support this, providers set a "change_id" field which groups the same change to multiple projects and environments so only one audit-log record is written in Sentry.

Reviewer: ```suggestion
Sentry does not currently have a concept of disambiguating flag changes by project and environment. Everything is done at the organization level. Flags which change for multiple environments or projects, or whatever internal grouping exists in the provider, must be de-duplicated. To support this, requests set a "change_id" field, which groups changes from multiple projects and environments so only one audit-log record is written in Sentry.
```
